python/sparkdl/graph_builder.py

```diff
@@ -0,0 +1,329 @@
+        """
+        Build a GraphFunction object
+
+        :param inputs: list, graph elements representing the inputs
+        :param outputs: list, graph elements representing the outputs
+        :param strip_and_freeze: bool, should we remove unused part of the graph and freee its values
+        """
+        if strip_and_freeze:
+            gdef = self.strip_and_freeze_until(outputs)
+        else:
+            gdef = self.graph.as_graph_def(add_shapes=True)
+        return GraphFunction(graph_def=gdef,
+                             input_names=[self.valid_input(i) for i in inputs],
+                             output_names=[self.valid_output(o) for o in outputs])
+
+    
+    def strip_and_freeze_until(self, fetches):
+        """
+        Converting all variables into constants
+        
+        :param fetches: list, graph elements representing the outputs of the graph
+        :return: GraphDef, the GraphDef object with cleanup procedure applied
+        """
+        gdef_frozen = tf.graph_util.convert_variables_to_constants(
+            self.sess, 
+            self.graph.as_graph_def(add_shapes=True),
+            [self.op_name(tnsr) for tnsr in fetches])
+        return gdef_frozen
+
+
+    def import_graph_function(self, gfn, input_map=None, name="GFN-IMPORT", **gdef_kargs):
+        """
+        Import a GraphFunction object into the current session
+        
+        :param gfn: GraphFunction, an object representing a TensorFlow graph and its inputs and outputs
+        :param input_map: dict, mapping from input names to existing graph elements 
+        :param name: str, the scope for all the variables in the GraphFunction's elements
+        :param gdef_kargs: other keyword elements for TensorFlow's `import_graph_def`
+        """
+        try:
+            del gdef_kargs["return_elements"]
+        except KeyError: 
+            pass
+        if input_map is not None:
+            assert set(input_map.keys()) <= set(gfn.input_names), \
+                "cannot locate provided input elements in the graph"
+
+        input_names = gfn.input_names
+        output_names = gfn.output_names
+        if name is not None:
+            name = name.strip()
+            if len(name) > 0:
+                output_names = [
+                    name + '/' + op_name for op_name in gfn.output_names]
+                input_names = [
+                    name + '/' + op_name for op_name in gfn.input_names]
+            
+        # When importing, provide the original output op names
+        tf.import_graph_def(gfn.graph_def,
+                            input_map=input_map,
+                            return_elements=gfn.output_names,
+                            name=name,
+                            **gdef_kargs)
+        feeds = list(map(self.get_tensor, input_names))
+        fetches = list(map(self.get_tensor, output_names))
+        return (feeds, fetches)
+
+
+class GraphBuilderSession(object):
```

Comment:
`IsolatedSession` might be a good name to describe what it's doing. It should match what tf.Session is (including being able to get the graph from the session), but ensure isolation for tensorflow and keras sessions.